### PR TITLE
Set noindex for preview builds of docs site

### LINKED ISSUE #28753: Some versions of docs are being indexed by browsers
# Improve documentation

image

https://docs-rog1zs1kv-supabase.vercel.app/docs/guides/getting-started/tutorials/with-vue-3

docs page

## Describe the problem

Some deploys with temporal subdomains are being indexed by browsers that can lead to old docs

## Describe the improvement

Avoid temporal subdomains being indexed by robots or somthing like that

### PR DESCRIPTION
**Pull request by encima**
## I have read the [CONTRIBUTING.md](https://github.com/supabase/supabase/blob/master/CONTRIBUTING.md) file.

YES

## What kind of change does this PR introduce?

Bug fix

## What is the current behavior?

Currently, preview builds of the site (i.e. https://docs-rog1zs1kv-supabase.vercel.app) are indexed as `X-Robots-Tag` is set to `all`. By default, [Vercel](https://vercel.com/guides/are-vercel-preview-deployment-indexed-by-search-engines) sets this to `noindex` and we override it here

## What is the new behavior?

This adds an additional check for the host domain and sets the header to `noindex` when it is not the production site

## Additional context

Fixes #28753

### CODE REVIEW

**Review by alaister — APPROVED** [review]
LGTM!

- confirmed that the `x-robots-tag: noindex` appears on the staging url

I guess we'll just have to confirm it's not there once this is in prod?

File: apps/docs/next.config.mjs [hunk 1]
```diff
@@ -111,6 +111,27 @@ const nextConfig = {
...
+          {
+            type: 'host',
+            value: 'supabase.com',
+          },
+        ],
+      },
+      {
+        source: '/:path*',
+        headers: [
+          {
```
Comment (charislam) [inline apps/docs/next.config.mjs:124]: Do we want to preserve the other headers here? (sts and x-frame-options)

Comment (encima) [reply apps/docs/next.config.mjs:124]: Up to you. No strong opinions and I thought things already merged 😅

Comment (charislam) [reply apps/docs/next.config.mjs:124]: haha 😄 cool I'll add them in and merge

**Pull request merged**